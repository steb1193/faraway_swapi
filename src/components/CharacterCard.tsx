'use client';
import type { FC } from 'react';
import {
  Card,
  CardActionArea,
  CardContent,
  Typography,
  Box,
  Skeleton,
} from '@mui/material';
import { CharacterDescription } from '@/api/types';
import { useRouter } from 'next/navigation';
import { getIdFromUrlString } from '@/helpers/getIdFromUrlString';

export type CharacterCardProps = Partial<CharacterDescription> & {
  disabled: boolean;
};

export const CharacterCard: FC<CharacterCardProps> = ({
  name,
  height,
  mass,
  hair_color,
  skin_color,
  eye_color,
  birth_year,
  gender,
  url,
  disabled,
}) => {
  const router = useRouter();

  const enhancedColor = (eye_color || '').includes('-')
    ? `color-mix(in lch, ${(eye_color || '').replaceAll('-', ', ')})`
    : eye_color;
  const handleClick = () => {
    if (!url) return;
    const id = getIdFromUrlString(url);
    router.push(`/${id}`);
  };
  if (disabled) {
    return (
      <>
        <Skeleton variant="rectangular" width={'100%'} height={118} />
        <Box sx={{ pt: 0.5 }}>
          <Skeleton />
          <br />
          <Skeleton width="50%" />
          <Skeleton width="40%" />
          <Skeleton width="70%" />
          <Skeleton width="70%" />
          <Skeleton width="60%" />
          <Skeleton width="40%" />
        </Box>
      </>
    );
  }
  return (
    <Card sx={{ maxWidth: 345 }} onClick={handleClick}>
      <CardActionArea>
        <Box height={140} bgcolor={enhancedColor} />
        <CardContent>
          <Typography gutterBottom variant="h6" component="div">
            {name || ''}
          </Typography>
          <Typography variant="body2" color="text.secondary">
            height: {height};
            <br />
            mass: {mass};
            <br />
            hair color: {hair_color};
            <br />
            skin color: {skin_color};
            <br />
            eye color: {eye_color};
            <br />
            birth year: {birth_year};
            <br />
            gender: {gender};
          </Typography>
        </CardContent>
      </CardActionArea>
    </Card>
  );
};
