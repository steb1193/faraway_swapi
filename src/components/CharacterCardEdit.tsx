'use client';
import type { ChangeEvent, FC } from 'react';
import {
  Card,
  CardActions,
  CardContent,
  Typography,
  Button,
  Box,
  TextField,
} from '@mui/material';
import { CharacterDescription } from '@/api/types';
import { useState } from 'react';
import Cookies from 'js-cookie';

export type CharacterCardEdit = Partial<CharacterDescription>;

export const CharacterCardEdit: FC<CharacterCardEdit> = ({
  name,
  height,
  mass,
  hair_color,
  skin_color,
  eye_color,
  birth_year,
  gender,
  url,
}) => {
  const getDefaultValue = (): Partial<CharacterDescription> => ({
    name,
    height,
    mass,
    hair_color,
    skin_color,
    eye_color,
    birth_year,
    gender,
    url,
  });
  const localData = Cookies.get(url!);
  const [formState, setFormState] = useState<Partial<CharacterDescription>>(
    localData ? JSON.parse(localData) : getDefaultValue(),
  );
  const enhancedColor = (formState.eye_color || '').includes('-')
    ? `color-mix(in lch, ${(formState.eye_color || '').replaceAll('-', ', ')})`
    : formState.eye_color;
  const handleReset = () => {
    if (!url) return;
    localStorage.removeItem(url);
    setFormState(getDefaultValue());
  };
  const handleEdit = (
    e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
    key: keyof CharacterCardEdit,
  ) => {
    if (!url) return;
    setFormState((prev) => {
      const newValue = { ...prev, [key]: e.target.value };
      Cookies.set(url!, JSON.stringify(newValue));
      return newValue;
    });
  };
  return (
    <Box
      minHeight="90vh"
      display="flex"
      justifyContent="center"
      alignItems="center"
    >
      <Card sx={{ width: 345 }}>
        <Box height={140} bgcolor={enhancedColor} />
        <CardContent>
          <Typography gutterBottom variant="h6" component="div">
            {name || ''}
          </Typography>
          <TextField
            fullWidth
            margin="dense"
            label="height"
            value={formState.height}
            onChange={(e) => handleEdit(e, 'height')}
            variant="standard"
          />
          <TextField
            fullWidth
            margin="dense"
            label="mass"
            value={formState.mass}
            onChange={(e) => handleEdit(e, 'mass')}
            variant="standard"
          />
          <TextField
            fullWidth
            margin="dense"
            label="hair color"
            value={formState.hair_color}
            onChange={(e) => handleEdit(e, 'hair_color')}
            variant="standard"
          />
          <TextField
            fullWidth
            margin="dense"
            label="skin color"
            value={formState.skin_color}
            onChange={(e) => handleEdit(e, 'skin_color')}
            variant="standard"
          />
          <TextField
            fullWidth
            margin="dense"
            label="eye color"
            value={formState.eye_color}
            onChange={(e) => handleEdit(e, 'eye_color')}
            variant="standard"
          />
          <TextField
            fullWidth
            margin="dense"
            label="birth year"
            value={formState.birth_year}
            onChange={(e) => handleEdit(e, 'birth_year')}
            variant="standard"
          />
          <TextField
            fullWidth
            margin="dense"
            label="gender"
            value={formState.gender}
            onChange={(e) => handleEdit(e, 'gender')}
            variant="standard"
          />
        </CardContent>

        <CardActions>
          <Button size="small" color="primary" onClick={handleReset}>
            Reset
          </Button>
        </CardActions>
      </Card>
    </Box>
  );
};
