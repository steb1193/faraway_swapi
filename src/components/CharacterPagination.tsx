'use client';
import { FC } from 'react';
import { useQuery } from '@tanstack/react-query';
import * as React from 'react';
import { getCharacters } from '@/api/getCharacters';
import { Pagination } from '@mui/material';
import { usePathname, useRouter, useSearchParams } from 'next/navigation';

export const CharacterPagination: FC = () => {
  const searchParams = useSearchParams();

  const router = useRouter();
  const pathname = usePathname();
  const { data, isFetching } = useQuery({
    queryKey: [
      'characters',
      { page: searchParams.get('page'), search: searchParams.get('search') },
    ],
    queryFn: async () => {
      const { data } = await getCharacters({
        page: searchParams.get('page') || undefined,
        search: searchParams.get('search') || undefined,
      });
      return data;
    },
    keepPreviousData: true,
    refetchOnWindowFocus: false,
    refetchOnReconnect: false,
    retry: false,
    staleTime: 300000,
  });

  const onChange: (
    event: React.ChangeEvent<unknown>,
    page: number,
  ) => void = async (e, page) => {
    if (`${page}` === searchParams.get('page')) {
      return;
    }
    const current = new URLSearchParams(Array.from(searchParams.entries()));
    if (!page) {
      current.delete('page');
    } else {
      current.set('page', `${page}`);
    }
    const search = current.toString();
    const query = search ? `?${search}` : '';
    const href = `${pathname}${query}`;
    router.push(href);
  };
  const pageCount = data?.count ? Math.ceil(data?.count / 10) : 0;
  if (pageCount <= 1) {
    return;
  }
  return (
    <Pagination
      disabled={isFetching}
      count={pageCount}
      page={+(searchParams.get('page') || 1)}
      onChange={onChange}
    />
  );
};
