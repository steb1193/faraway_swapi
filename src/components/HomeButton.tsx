'use client';
import { FC } from 'react';
import { useRouter } from 'next/navigation';
import HomeIcon from '@mui/icons-material/Home';
import { IconButton } from '@mui/material';

export const HomeButton: FC = () => {
  const router = useRouter();

  return (
    <IconButton
      size="large"
      edge="start"
      color="inherit"
      aria-label="open drawer"
      sx={{ mr: 2 }}
      onClick={() => router.push('/')}
    >
      <HomeIcon />
    </IconButton>
  );
};
