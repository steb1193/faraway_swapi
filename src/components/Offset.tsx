'use client';
import { styled } from '@mui/material/styles';
import { FC } from 'react';

export const Offset: FC = styled('div')(({ theme }) => theme.mixins.toolbar);
