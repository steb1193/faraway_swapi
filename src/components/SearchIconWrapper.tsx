'use client';
import { styled } from '@mui/material/styles';
import type { FC } from 'react';
import { PropsWithChildren } from 'react';

export const SearchIconWrapper: FC<PropsWithChildren> = styled('div')(
  ({ theme }) => ({
    padding: theme.spacing(0, 2),
    height: '100%',
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  }),
);
