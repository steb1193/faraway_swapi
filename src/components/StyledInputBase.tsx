'use client';
import { styled } from '@mui/material/styles';
import { InputBase, InputBaseProps, Theme } from '@mui/material';
import { StyledComponent } from '@emotion/styled';
import { MUIStyledCommonProps } from '@mui/system';

export const StyledInputBase: StyledComponent<
  InputBaseProps & MUIStyledCommonProps<Theme>
> = styled(InputBase)(({ theme }) => ({
  color: 'inherit',
  '& .MuiInputBase-input': {
    padding: theme.spacing(1, 1, 1, 0),
    // vertical padding + font size from searchIcon
    paddingLeft: `calc(1em + ${theme.spacing(4)})`,
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      width: '12ch',
      '&:focus': {
        width: '20ch',
      },
    },
  },
}));
