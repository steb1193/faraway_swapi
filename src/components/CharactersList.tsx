'use client';
import { FC } from 'react';
import { CharacterCard } from '@/components/CharacterCard';
import { Grid } from '@mui/material';
import { useGetCharacters } from '@/hooks/useGetCharacters';

export const CharactersList: FC = () => {
  const { data, isFetching } = useGetCharacters();

  return (
    <Grid
      container
      spacing={{ xs: 2, md: 3 }}
      columns={{ xs: 4, sm: 8, md: 10 }}
    >
      {data?.results?.map((character, key) => (
        <Grid item xs={4} sm={4} md={2} key={character.url + key}>
          <CharacterCard {...character} disabled={isFetching} />
        </Grid>
      ))}
    </Grid>
  );
};
