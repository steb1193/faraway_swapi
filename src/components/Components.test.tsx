import { render } from '@testing-library/react';
import mockRes from '@/__mocks__/people.json';
import { CharacterCard } from '@/components/CharacterCard';
import { CharacterCardEdit } from '@/components/CharacterCardEdit';
import { HomeButton } from '@/components/HomeButton';
import { Offset } from '@/components/Offset';
import { Search } from '@/components/Search';
import { SearchIconWrapper } from '@/components/SearchIconWrapper';
import { StyledInputBase } from '@/components/StyledInputBase';

describe('Components', () => {
  beforeEach(() => {
    global['mockAxios'].onGet('people').reply(200, mockRes);
    global['mockAxios'].onGet(/people\/*/).reply(200, mockRes.results[0]);
  });

  test('renders CharacterCard enabled unchanged', () => {
    const { container } = render(
      <CharacterCard {...mockRes.results[0]} disabled={false} />,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders CharacterCard disabled unchanged', () => {
    const { container } = render(
      <CharacterCard {...mockRes.results[0]} disabled={true} />,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders CharacterCardEdit unchanged', () => {
    const { container } = render(<CharacterCardEdit {...mockRes.results[0]} />);
    expect(container).toMatchSnapshot();
  });

  test('renders HomeButton unchanged', () => {
    const { container } = render(<HomeButton />);
    expect(container).toMatchSnapshot();
  });

  test('renders Offset unchanged', () => {
    const { container } = render(<Offset />);
    expect(container).toMatchSnapshot();
  });

  test('renders Search unchanged', () => {
    const { container } = render(<Search />);
    expect(container).toMatchSnapshot();
  });

  test('renders SearchIconWrapper unchanged', () => {
    const { container } = render(<SearchIconWrapper />);
    expect(container).toMatchSnapshot();
  });

  test('renders StyledInputBase unchanged', () => {
    const { container } = render(<StyledInputBase />);
    expect(container).toMatchSnapshot();
  });
});
