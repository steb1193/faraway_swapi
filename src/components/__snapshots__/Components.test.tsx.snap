// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components renders CharacterCard disabled unchanged 1`] = `
<div>
  <span
    class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-58ftfq-MuiSkeleton-root"
    style="width: 100%; height: 118px;"
  />
  <div
    class="MuiBox-root css-37urdo"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
    />
    <br />
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
      style="width: 50%;"
    />
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
      style="width: 40%;"
    />
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
      style="width: 70%;"
    />
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
      style="width: 70%;"
    />
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
      style="width: 60%;"
    />
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
      style="width: 40%;"
    />
  </div>
</div>
`;

exports[`Components renders CharacterCard enabled unchanged 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1ri6ub7-MuiPaper-root-MuiCard-root"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <div
        class="MuiBox-root css-ns1dge"
      />
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
        >
          Luke Skywalker
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          height: 
          172
          ;
          <br />
          mass: 
          77
          ;
          <br />
          hair color: 
          blond
          ;
          <br />
          skin color: 
          fair
          ;
          <br />
          eye color: 
          blue
          ;
          <br />
          birth year: 
          19BBY
          ;
          <br />
          gender: 
          male
          ;
        </p>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`Components renders CharacterCardEdit unchanged 1`] = `
<div>
  <div
    class="MuiBox-root css-gbtijt"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1syf4pq-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiBox-root css-ns1dge"
      />
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
        >
          Luke Skywalker
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r0:"
            id=":r0:-label"
          >
            height
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id=":r0:"
              type="text"
              value="172"
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r1:"
            id=":r1:-label"
          >
            mass
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id=":r1:"
              type="text"
              value="77"
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r2:"
            id=":r2:-label"
          >
            hair color
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id=":r2:"
              type="text"
              value="blond"
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r3:"
            id=":r3:-label"
          >
            skin color
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id=":r3:"
              type="text"
              value="fair"
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r4:"
            id=":r4:-label"
          >
            eye color
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id=":r4:"
              type="text"
              value="blue"
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r5:"
            id=":r5:-label"
          >
            birth year
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id=":r5:"
              type="text"
              value="19BBY"
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r6:"
            id=":r6:-label"
          >
            gender
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id=":r6:"
              type="text"
              value="male"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-1knaqv7-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Reset
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Components renders HomeButton unchanged 1`] = `
<div>
  <button
    aria-label="open drawer"
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="HomeIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
      />
    </svg>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`Components renders Offset unchanged 1`] = `
<div>
  <div
    class="css-1x551m1"
  />
</div>
`;

exports[`Components renders Search unchanged 1`] = `
<div>
  <div
    class="css-1orzy9s"
  />
</div>
`;

exports[`Components renders SearchIconWrapper unchanged 1`] = `
<div>
  <div
    class="css-1iz4iln"
  />
</div>
`;

exports[`Components renders StyledInputBase unchanged 1`] = `
<div>
  <div
    class="MuiInputBase-root MuiInputBase-colorPrimary css-7mt4h1-MuiInputBase-root"
  >
    <input
      class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
      type="text"
      value=""
    />
  </div>
</div>
`;
